apps/api/src/tasks/bank-import/import-transactions.task.ts

```diff
@@ -401,7 +509,19 @@ export class IrisTasks {
 
   private async saveBankTrxRecords(data: filteredTransaction[]) {
     // Insert new transactions
-    const inserted = await this.prisma.bankTransaction.createMany({ data, skipDuplicates: true })
+    const inserted = await this.prisma.$transaction(
+      data.map((trx) =>
+        this.prisma.bankTransaction.upsert({
+          where: { id: trx.id },
+          update: {
+            bankDonationStatus: trx.description.startsWith(this.AFFILIATE_CODE_PREFIX)
+              ? trx.bankDonationStatus
+              : undefined,
```

Comment:
wouldn't this set the status of existing non-affiliate donation to undefined, which would be a problem if it was marked imported already

Reply:
I will test it out just to be certain, but as per prisma's documentation `undefined` means **do nothing/ don't update** 
[Reference](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined)

Reply:
You are right. Than this is a completely fine